src/libraries/Microsoft.Extensions.Configuration.EnvironmentVariables/tests/EnvironmentVariablesTest.cs

```diff
@@ -153,6 +153,38 @@ public void ReplaceDoubleUnderscoreInEnvironmentVariables()
             Assert.Equal("System.Data.SqlClient", envConfigSrc.Get("ConnectionStrings:_db1_ProviderName"));
         }
 
+        [Fact]
+        public void ReplaceDoubleUnderscoreInEnvironmentVariablesButNotPrefix()
+        {
+            var dict = new Hashtable()
+                {
+                    {"test__prefix__with__double__underscores__data__ConnectionString", "connection"},
+                    {"SQLCONNSTR__db1", "connStr"}
+                };
+            var envConfigSrc = new EnvironmentVariablesConfigurationProvider("test__prefix__with__double__underscores__");
+
+            envConfigSrc.Load(dict);
+
+            Assert.Equal("connection", envConfigSrc.Get("data:ConnectionString"));
+            Assert.Equal("System.Data.SqlClient", envConfigSrc.Get("ConnectionStrings:_db1_ProviderName"));
```

Comment:
Could you please also add a test to cover the example in the comment below:

https://github.com/dotnet/runtime/issues/40911#issuecomment-686805871

Reply:
I will